generic_config_updater/patch_sorter.py

```diff
@@ -0,0 +1,1012 @@
+import copy
+import json
+import jsonpatch
+from collections import deque
+from enum import Enum
+from .gu_common import OperationWrapper, OperationType, GenericConfigUpdaterError, JsonChange, PathAddressing
+
+# from gu_common import OperationWrapper, OperationType, GenericConfigUpdaterError, JsonChange, PathAddressing
+
+class Diff:
+    """
+    A class that contains the diff info between current and target configs. 
+    """
+    def __init__(self, current_config, target_config):
+        self.current_config = current_config
+        self.target_config = target_config
+
+    def __hash__(self):
+        cc = json.dumps(self.current_config, sort_keys=True)
+        tc = json.dumps(self.target_config, sort_keys=True)
+        return hash(cc) ^ hash(tc)
```

Comment:
>hash(cc) ^ hash(tc) [](http://example.com/codeflow?start=15&length=19)

How about `hash ((cc, tc))`?
The benefit is swapping input will result in different value.

Reply:
good idea. updated